aws-storage-s3/src/test/java/com/amplifyframework/storage/s3/operation/AWSS3StorageGetPresignedUrlOperationTest.kt

```diff
@@ -138,4 +139,81 @@ public class AWSS3StorageGetPresignedUrlOperationTest {
         awsS3StorageGetPresignedUrlOperation.start()
         Mockito.verify(storageService).getPresignedUrl(expectedKey, 1, false)
     }
+
+    @Test
+    fun `getPresignedUrl fails with non existent S3 path when validateObjectExistence is enabled`() {
```

Comment:
this would be a good E2E test to have instead of using mock storage service, we would just need to create a random path so it's non existent and pass in the option, right?

Reply:
yes, E2E were added for bot positive/negative cases, see `AWSS3StoragePathGetUrlTest.kt`